src/python/pants/backend/scala/test/scalatest.py

```diff
@@ -57,42 +78,61 @@ async def run_scalatest_test(
     }
 
     reports_dir_prefix = "__reports_dir"
-    reports_dir = f"{reports_dir_prefix}/{field_set.address.path_safe_spec}"
+    reports_dir = f"{reports_dir_prefix}/{request.field_set.address.path_safe_spec}"
 
     # Classfiles produced by the root `scalatest_test` targets are the only ones which should run.
     user_classpath_arg = ":".join(classpath.root_args())
 
-    process_result = await Get(
-        FallibleProcessResult,
-        Process(
-            argv=[
-                *jdk_setup.args(
-                    bash,
-                    [
-                        *classpath.args(),
-                        *scalatest_classpath.classpath_entries(toolcp_relpath),
-                    ],
-                ),
-                "org.scalatest.tools.Runner",
-                # TODO: We currently give the entire user classpath to the JVM for startup (which
-                # mixes it with the user classpath), and then only specify the roots to run here.
-                #   see https://github.com/pantsbuild/pants/issues/13871
-                *(("-R", user_classpath_arg) if user_classpath_arg else ()),
-                "-o",
-                "-u",
-                reports_dir,
-                *scalatest.options.args,
-            ],
-            input_digest=merged_classpath_digest,
-            immutable_input_digests=immutable_input_digests,
-            output_directories=(reports_dir,),
-            append_only_caches=jdk_setup.append_only_caches,
-            env=jdk_setup.env,
-            description=f"Run Scalatest runner for {field_set.address}",
-            level=LogLevel.DEBUG,
-        ),
+    # Cache test runs only if they are successful, or not at all if `--test-force`.
+    cache_scope = (
+        ProcessCacheScope.PER_SESSION if test_subsystem.force else ProcessCacheScope.SUCCESSFUL
     )
 
+    extra_jvm_args: list[str] = []
```

Comment:
Should this default to the value of a `--scalatest-jvm-args` option? This would allow setting JVM memory options for tests regardless of whether `--test-debug` was in effect.

Reply:
Let's hold off on adding JVM options until #13067: they'll need standardized templating of some sort.